Functional, just need to integrate it with backend now

Noel-Alex opened the pull request.

Noel-Alex committed:
build: moved vector store to the cloud

Noel-Alex committed:
docs: added docs for summaries

Noel-Alex committed:
refactor: changed requirements.txt to remove unnecessary modules

Noel-Alex committed:
feat: Added the summary_clear command

Noel-Alex committed:
merge commit

Noel-Alex committed:
refactor: removed unnecessary modules

Noel-Alex committed:
refactor: got rid of unecessary imports

Noel-Alex committed:
Merged main into dev

Noel-Alex committed:
uncommented authorisatiion check in summary.py

Noel-Alex committed:
uncommented authorisatiion check in summary.py

Noel-Alex committed:
MERGE COMMIT

Noel-Alex committed:
removed pinecone vector db

Noel-Alex committed:
fix: fixed some bugs with Groq api calls

Noel-Alex committed:
MERGE COMMIT

Noel-Alex committed:
Merge branch 'sidhant-sriv:main' into main

Noel-Alex committed:
feat: databse.py created containing functions to add summaries to a server through ultra-chat-backend

Noel-Alex committed:
feat: Moved all summary stores to the server via ultra-chat-backend

Noel-Alex committed:
Merge branch 'main' of https://github.com/Noel-Alex/ultrachat

Noel-Alex committed:
chore: updated requirements.txt

sidhant-sriv commented on Dockerfile:
```diff
@@ -1,17 +1,17 @@
+COPY . .
+
+# Specify the command to run the application
+CMD ["python", "app.py"]
```
Use gunicorn here

sidhant-sriv commented on app.py:
```diff
@@ -1,68 +1,68 @@
+intents.message_content = True  # NOQA
+bot = commands.Bot(command_prefix='!', intents=intents)
+
+EXCLUDED_FILES = ['music.py', 'summarise.py'] #files that might be in the cogs folder but need not to be loaded
```
delete whatever isn't needed

sidhant-sriv commented on cogs/summaries.py:
```diff
@@ -64,31 +63,104 @@ async def button_callback1(self, interaction: discord.Interaction, button: disco
+                            view=SummariesView(pages=pages, num=num, summary_ids=summary_ids, id = ctx.author.id))
+
+    '''@commands.command(name="summary_clear")
+    async def summary_clear(self,ctx, start: int =0, stop = None):
+        start = int(start)
+        stop = int(stop) if stop is not None else None
+        summary_folder = f'chats/{ctx.author.name}/{ctx.message.guild.name}/summaries'
+        if (start>stop) and (start>0 and stop<len(os.listdir(summary_folder))):
+            if stop is not None:
+                for filename in os.listdir(summary_folder)[start-1:stop-1]:
+                    summary = os.path.join(summary_folder, filename)
+                    os.remove(summary)
+
+        if stop is None and start > 0:
+            os.remove(os.path.join(summary_folder, f"{start}.txt"))
 
-#    @commands.command(name="summaries_clear"):
+        i = 1
+        for filename in os.listdir(summary_folder):
+            old_summary = os.path.join(summary_folder, filename)
+            new_summary = os.path.join(summary_folder, f"{i}.txt")
+            os.rename(old_summary, new_summary)
+            i += 1'''
```
Fix

sidhant-sriv commented on cogs/summary.py:
```diff
@@ -144,16 +149,17 @@ async def is_authenticated(self, user_id):
+        login_embed.set_footer(text="Click the link above to login.")
 
-        login_url = "https://discord.com/oauth2/authorize?client_id=1256967412943949904&redirect_uri=https://ultra-chat-backend.onrender.com/callback&response_type=code&scope=identify%20email"
+        login_url = "https://discord.com/api/oauth2/authorize?client_id=1256967412943949904\u0026redirect_uri=https://ultra-achat-go-backend.onrender.com/callback\u0026response_type=code\u0026scope=identify%20email"
```
Don't hardcode

sidhant-sriv commented on docker-compose.yml:
```diff
@@ -1,7 +1,7 @@
+  discord-bot:
+    build:
+      context: .
+      dockerfile: Dockerfile # Replace with your actual Dockerfile name if different
```
this is an optional param

sidhant-sriv reviewed (commented):
Use a formatter like black to clean up code. Remove unnecessary comments and prints.

Noel-Alex committed:
doc: adding documentation for the bot and misc stuff

Noel-Alex committed:
feat: Switched to Cohere embeddings, added prompt template for summary engine and added README

Noel-Alex committed:
docs: fixed README

Noel-Alex committed:
docs: started on commenting within code
docker compose update incomplete, added folders generated by code to dockerignore, updated env.example to hold login url and cohere key

Noel-Alex committed:
Merge branch 'dev'

Noel-Alex committed:
docs: Updated README
